src/resources/resolver.js

```diff
@@ -0,0 +1,113 @@
+import _ from 'lodash';
+import UriTemplate from 'urijs/src/URITemplate';
+import { getStatus } from '../status';
+import { RESOLVED_ENDPOINT } from '../consts';
+import rio from '../rio';
+import { validateResourceConfig } from './validation';
+
+export function resolveSchemaType(config) {
+  const schemaConfig = _.get(config, 'schema');
+
+  if (_.isString(schemaConfig)) {
+    return schemaConfig;
+  }
+
+  return _.get(schemaConfig, 'type');
+}
+
+export function resolveSchema(config) {
+  const schemaConfig = _.get(config, 'schema');
+
+  if (_.isString(schemaConfig)) {
+    return null;
+  }
+
+  return schemaConfig;
+}
+
+function resolveAction(config, action) {
+  const actionConfig = _.get(config, ['actions', action], {});
+
+  return _
+    .chain({})
```

Comment:
Maybe not that important, but interesting: https://medium.com/making-internets/why-using-chain-is-a-mistake-9bc1f80d51ba

Reply:
Thanks, very useful article.